cpp/src/io/parquet/page_data.cu

```diff
@@ -959,7 +981,16 @@ static __device__ bool setupLocalPageInfo(page_state_s* const s,
           [[fallthrough]];
         case DOUBLE: s->dtype_len = 8; break;
         case INT96: s->dtype_len = 12; break;
-        case BYTE_ARRAY: s->dtype_len = sizeof(string_index_pair); break;
+        case BYTE_ARRAY:
+          if (s->col.converted_type == DECIMAL) {
+            auto const decimal_precision = s->col.decimal_precision;
+            s->dtype_len = decimal_precision <= MAX_DECIMAL32_PRECISION   ? sizeof(int32_t)
+                           : decimal_precision <= MAX_DECIMAL64_PRECISION ? sizeof(int64_t)
+                                                                          : sizeof(__int128_t);
```

Comment:
Would recommend using if-statements instead of nested ternaries like
https://github.com/rapidsai/cudf/pull/12205/files#diff-d42b9912ad4c2fb482d2c15ee3eb2d6935d878a17a436f7d39aa8c0ab2dcedc5R122
and
https://github.com/rapidsai/cudf/pull/12205/files#diff-672b345ca18d5957ee39bd7802bbf67f454cb941ab78174fbab3dc1ddaa048b3R1867

Reply:
Good point. Done